### DISCUSSION TITLE
Creating a new workflow definition will not populate createTime

### DISCUSSION THREAD

**Discussion by buzz08** [Netflix/conductor#2625] [General]
**Describe the bug**
When you create a new workflow definition, it doesn't set the create time. This causes issues in the workflow definitions UI page, especially if you try to sort by create time. We need to set this value as a current time when you create the workflow.

**Details**
Conductor version: Latest
Persistence implementation: NA
Queue implementation: NA
Lock:  NA
Workflow definition: Applicable to new workflows
Task definition: NA
Event handler definition: NA

**To Reproduce**
1. Create a new workflow without createTime or updateTime
2. Retrieve the workflow definition, it won't have createTime

**Expected behavior**
createTime should be defaulted to the system time

**Screenshots**
NA

**Additional context**
NA

**Comment by jxu-nflx** [comment]
Hello @buzz08 Do you want to submit a PR to fix this? Thank you!

**Comment by aravindanr** [comment]
I was able to reproduce it. PUT /workflow does not set the `createTime`, but POST /workflow defaults it to the current server time.

**Comment by apanicker-nflx** [comment]
Since we are altering the metadata APIs, we propose that we should make these consistent. 
For example, the task definition APIs allow creation of an array of task definitions but update of one task definition. Further, the update operation is only allowed if the task definition has already been registered.

The proposal is to follow the same convention for workflow definition APIs as well. 
As part of the PR #2617, each update operation is now a read-before-write (getWorkflowDef before update). Further, the create API allows only one workflow definition but the update API allows an array of workflow definitions. Here again, the update workflow definition should only be allowed if the workflow definition has already been registered with the system.

API consistency across both workflow and task definitions would provide for a better user/developer experience and reduce support burden.
